docs: Mention privacy implications of reprovider settings (#9207)
ipfs/kubo: docs/config.md

Tells reprovider what should be announced. Valid strategies are:

> *Some options have better privacy than others.*
> 

Reviewer: ```suggestion
```

- `"all"` - announce all CIDs of stored blocks
- `"pinned"` - only announce pinned CIDs recursively (both roots and child blocks)

> - `"roots"` - only announce the root block of explicitly pinned CIDs (offers the best privacy)

Reviewer: ```suggestion
- `"roots"` - only announce the root block of explicitly pinned CIDs
```

> **DISCLAIMER: Assume that your CIDs are public information regardless of the option you choose.**

Reviewer: ```suggestion
Kubo assumes that CIDs are public information. For example they are publicly announced in the network DHT.
```

Tells reprovider what should be announced. Valid strategies are:

*Some options have better privacy than others.*

- `"all"` - announce all CIDs of stored blocks
- `"pinned"` - only announce pinned CIDs recursively (both roots and child blocks)
- `"roots"` - only announce the root block of explicitly pinned CIDs (offers the best privacy)

**DISCLAIMER: Assume that your CIDs are public information regardless of the option you choose.**

> Learn more on [IPFS Docs](https://docs.ipfs.tech/concepts/privacy-and-encryption/)

Reviewer 2: ```suggestion
Learn more about privacy and encryption on [IPFS Docs](https://docs.ipfs.tech/concepts/privacy-and-encryption/)
```
